src/main/java/com/bravo/user/service/UserService.java

```diff
@@ -40,6 +41,24 @@ public UserReadDto create(final UserSaveDto request){
     return resourceMapper.convertUser(user);
   }
 
+  public List<UserReadDto> retrieveByName(
+          final String name,
+          final PageRequest pageRequest,
+          final HttpServletResponse httpResponse){
+    UserFilter filter = new UserFilter();
+    String[] names = name.split(" ");
```

Comment:
This is a straight-forward way to parse a name with spaces.  This forces a first name to be passed in though.  Is there a way to make this so it can search by any part of a name?

Reply:
Yes, this could be done several ways. If the desire was to create a search for the name value in any of the fields, I would have provided and OR query implementation. The way I interpreted the requirement though, was that a single name value was provided to be able to search for a full or partial value within the name.